src/grpc-virtual-machine.cpp

```diff
@@ -348,8 +348,9 @@ interpreter_break_reason grpc_virtual_machine::do_run(uint64_t mcycle_end) {
         return interpreter_break_reason::yielded_manually;
     } else if (response.iflags_x()) {
         return interpreter_break_reason::yielded_automatically;
+    } else if (response.mcycle() != mcycle_end) {
```

Comment:
Hmmm. Should we give preference to the soft yield over mcycle_end? This way, the program controlling the emulator in a loop would receive the `yielded_softly`, deal with it, restart the emulator, and only then get the `reached_target_mcycle`. Otherwise, the soft yield is lost.

Reply:
There is no flag for this soft yield. So, unlike the other yields, there is no other way for the program controlling the emulator to know.

Reply:
Good catch, indeed the way I did soft yields are lost in case it yields exactly at `mcycle_end`, will fix this.

Reply:
So, I did this comparing `response.mcycle() != mcycle_end` because I was trying to avoid doing changes in grpc interfaces. Now I did the proper way by forwarding from the run command, but I had to create a PR on grpc interfaces https://github.com/cartesi/grpc-interfaces/pull/28